drivers/LVHDoISCSISR.py

```diff
@@ -492,6 +492,9 @@ def delete(self, sr_uuid):
     def attach(self, sr_uuid):
         try:
             connected = False
+            if not self.iscsiSRs:
```

Comment:
instead of `if not self.iscsiSRs:` you might  have better results with `if len(self.iscsiSRs) == 0:`.

It could also be confused by the `try: except: else:`  construct to set `connected=True`, that could be moved into the main block and the `else:` removed.